### PR TITLE
chore: remove unused code

### PR DESCRIPTION
**Pull request by jmetev1**
Checklist:

- [x ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x ] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #none 

removing unused files and exports flagged by knip.

### CODE REVIEW

File: client/src/components/seo/index.tsx [hunk 1]
```diff
@@ -100,7 +100,7 @@ SEO.displayName = 'SEO';
 
 export default SEO;
 
-export const query = graphql`
```
Comment (ojeytonwilliams) [inline client/src/components/seo/index.tsx:103]: To make it a little more obvious that this isn't a page query, could you inline it?

File: client/utils/challenge-types.js [hunk 2]
```diff
@@ -25,7 +25,6 @@ exports.backEndProject = backEndProject;
 exports.pythonProject = pythonProject;
 exports.codeAllyCert = codeAllyCert;
 exports.colab = colab;
-exports.exam = exam;
```
Comment (ojeytonwilliams) [inline client/utils/challenge-types.js:28]: This is a work in progress, so should not be removed.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for extracting these changes, @jmetev1!

I spotted a few issues:

File: client/src/components/seo/index.tsx (hunk 1 above)
Comment (jmetev1) [reply client/src/components/seo/index.tsx:103]: sure thing

File: client/src/components/search/redux/index.js [hunk 3]
```diff
@@ -23,13 +23,11 @@ const types = createTypes(
 
 export const toggleSearchDropdown = createAction(types.toggleSearchDropdown);
 export const toggleSearchFocused = createAction(types.toggleSearchFocused);
-export const updateSearchIndexName = createAction(types.updateSearchIndexName);
```
[the remark this answers is not in the document]
Comment (jmetev1) [reply client/src/components/search/redux/index.js:26]: ah good point

File: client/src/components/Donation/paypal-button-script-loader.tsx [hunk 4]
```diff
@@ -76,19 +76,6 @@ export default class PayPalButtonScriptLoader extends Component<
...
-    props: PayPalButtonScriptLoaderProps,
-    state: PayPalButtonScriptLoaderState
-  ): { isSubscription: boolean } | null {
-    const { isSubscription } = props;
-    if (isSubscription !== state.isSubscription) {
-      return { isSubscription: isSubscription };
-    }
-    return null;
-  }
-
```
[the remark this answers is not in the document]
Comment (jmetev1) [reply client/src/components/Donation/paypal-button-script-loader.tsx]: whoops i can't believe knip flagged that and i can't believe i took it out. sorry!

**Review by ojeytonwilliams — APPROVED** [review]
Great, thanks!

LGTM :+1:

**Review by Sboonny — APPROVED** [review]
LGTM.

**Review by Sboonny — APPROVED** [review]
LGTM.

**Pull request merged**